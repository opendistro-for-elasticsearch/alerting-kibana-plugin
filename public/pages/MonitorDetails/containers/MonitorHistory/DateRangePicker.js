/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { EuiFlexGroup, EuiFlexItem, EuiDatePickerRange, EuiDatePicker } from '@elastic/eui';
import { MAX_DAYS_ALLOWED_IN_RANGE } from './utils/constants';
import { isToday, getRangeMaxTime } from './utils/timeUtils';

class DateRangePicker extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      rangeStartDateTime: {
        ...this.getStartDatePickerProps(props.initialStartTime, props.initialStartTime),
      },
      rangeEndDateTime: {
        ...this.getEndDatePickerProps(props.initialStartTime, props.initialEndTime),
      },
    };
  }

  getStartDatePickerProps = startDateTime => {
    return {
      selected: startDateTime.isAfter(moment()) ? moment().startOf('day') : startDateTime,
      minTime: startDateTime.clone().startOf('day'),
      maxTime: getRangeMaxTime(startDateTime),
      maxDate: moment(),
    };
  };

  getEndDatePickerProps = (startDateTime, endDateTime) => {
    const maxDate = startDateTime.clone().add(MAX_DAYS_ALLOWED_IN_RANGE, 'days');
    const now = moment();
    return {
      selected: endDateTime,
      minTime: endDateTime.clone().startOf('day'),
      maxTime: getRangeMaxTime(endDateTime),
      maxDate: maxDate > now ? now : maxDate,
      injectTimes: isToday(endDateTime) ? [now] : [],
    };
  };

  handleChangeStart = startDateTime => {
    const { onRangeChange } = this.props;
    this.setState(state => {
      // If preselected endTime is out of range / grater than start update end time accordingly
      let endTime = state.rangeEndDateTime.selected;
      if (endTime.isSame(startDateTime)) {
        endTime = startDateTime.clone().add(30, 'm');
      } else if (
        endTime.isBefore(startDateTime) ||
        endTime.diff(startDateTime, 'days') > MAX_DAYS_ALLOWED_IN_RANGE
      ) {
        endTime = startDateTime.clone().add(1, 'd');
      }
      onRangeChange(startDateTime, endTime);
      return {
        rangeStartDateTime: {
          ...this.getStartDatePickerProps(startDateTime, endTime),
        },
        rangeEndDateTime: {
          ...this.getEndDatePickerProps(startDateTime, endTime),
        },
      };
    });
  };

  handleChangeEnd = endDateTime => {
    const { onRangeChange } = this.props;
    this.setState(({ rangeStartDateTime, rangeEndDateTime }) => {
      let startTime = rangeStartDateTime.selected;
      if (startTime.isSame(endDateTime)) {
        startTime = endDateTime.clone().subtract(30, 'm');
      } else if (startTime.isAfter(endDateTime)) {
        startTime = endDateTime.clone().subtract(30, 'm');
      }
      onRangeChange(startTime, endDateTime);
      return {
        rangeStartDateTime: {
          ...rangeStartDateTime,
          selected: startTime,
        },
        rangeEndDateTime: {
          ...rangeEndDateTime,
          selected: endDateTime,
        },
      };
    });
  };
  render() {
    const { rangeStartDateTime, rangeEndDateTime } = this.state;
    return (
      <EuiFlexGroup alignItems="center" justifyContent="flexEnd">
        <EuiFlexItem grow={false} style={{ padding: '0 10px' }}>
          <EuiDatePickerRange
            fullWidth
            startDateControl={
              <EuiDatePicker
                selected={rangeStartDateTime.selected}
                onChange={this.handleChangeStart}
                startDate={rangeStartDateTime.selected}
                endDate={rangeEndDateTime.selected}
                isInvalid={rangeStartDateTime.selected > rangeStartDateTime.selected}
                aria-label="Start date"
                showTimeSelect
                popperClassName="euiRangePicker--popper"
                shouldCloseOnSelect
                {...rangeStartDateTime}
              />
            }
            endDateControl={
              <EuiDatePicker
                selected={rangeEndDateTime.selected}
                onChange={this.handleChangeEnd}
                startDate={rangeStartDateTime.selected}
                endDate={rangeEndDateTime.selected}
                isInvalid={rangeStartDateTime.selected > rangeStartDateTime.selected}
                aria-label="End date"
                showTimeSelect
                popperClassName="euiRangePicker--popper"
                shouldCloseOnSelect
                {...rangeEndDateTime}
              />
            }
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }
}

DateRangePicker.propTypes = {
  onRangeChange: PropTypes.func.isRequired,
};

export default DateRangePicker;
