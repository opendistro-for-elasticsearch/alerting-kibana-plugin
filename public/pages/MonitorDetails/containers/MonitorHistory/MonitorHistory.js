/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { get, isEqual } from 'lodash';
import chrome from 'ui/chrome';
import { EuiHorizontalRule } from '@elastic/eui';
import moment from 'moment';

import ContentPanel from '../../../../components/ContentPanel';
import {
  TriggersTimeSeries,
  POIChart,
  Legend,
  EmptyHistory,
} from '../../components/MonitorHistory/';
import { calculateInterval } from './utils/timeUtils';
import DateRangePicker from './DateRangePicker';

import {
  generateFirstDataPoints,
  dataPointsGenerator,
  getTimeSeriesSearchRequest,
  getPOISearchQuery,
} from './utils/chartHelpers';
import * as HistoryConstants from './utils/constants';
import { INDEX } from '../../../../../utils/constants';

class MonitorHistory extends PureComponent {
  constructor(props) {
    super(props);

    //TODO:: we might move this to DateRangePicker and make callback to update POIInterest? Check the perf and timings.
    this.initialStartTime = moment(Date.now())
      .subtract(HistoryConstants.DEFAULT_POI_TIME_WINDOW_DAYS, 'days')
      .startOf('day');
    this.initialEndTime = moment(Date.now());

    this.state = {
      poiData: [],
      triggersData: {},
      isLoading: true,
      maxAlerts: 0,
      timeSeriesWindow: {},
      poiTimeWindow: {
        startTime: this.initialStartTime,
        endTime: this.initialEndTime,
      },
    };
    this.isDarkMode = chrome.getUiSettingsClient().get('theme:darkMode') || false;
  }
  async componentDidMount() {
    const { triggers } = this.props;
    if (triggers.length > 0) {
      await this.getPOIData();
    }
  }

  componentDidUpdate(prevProps) {
    if (!isEqual(this.props.triggers, prevProps.triggers)) {
      this.getPOIData();
    }
  }

  handleRangeChange = (startTime, endTime) => {
    this.setState({ poiTimeWindow: { startTime, endTime } }, () => this.getPOIData());
  };

  /* This will generate data points and returns array
    {
      x0: plotting starts from.
      x: plotting ends,
      meta: Any meta data
  */
  generatePlotData = alertsData => {
    const {
      timeSeriesWindow: { startTime: windowStartTime, endTime: windowEndTime },
    } = this.state;
    // In case of empty just return  start-end as not alerting state.
    if (alertsData.length === 0) {
      return [
        {
          x0: windowStartTime,
          x: windowEndTime,
          state: HistoryConstants.TIME_SERIES_ALERT_STATE.NO_ALERTS,
          meta: {
            startTime: windowStartTime,
            endTime: windowEndTime,
          },
        },
      ];
    }
    const [firstAlert, ...restAlerts] = alertsData;
    //handles first alert separately.
    const firstAlertDataPoints = generateFirstDataPoints({
      startTime: get(firstAlert, '_source.start_time'),
      acknowledgedTime: get(firstAlert, '_source.acknowledged_time'),
      endTime: get(firstAlert, '_source.end_time'),
      state: get(firstAlert, '_source.state'),
      windowStartTime,
      windowEndTime,
      errorsCount: get(firstAlert, '_source.alert_history.length', 0),
    });
    // Handle rest of the alerts
    let lastEndTime = get(firstAlert, '_source.end_time');

    const restAlertsDataPoints = restAlerts.reduce((acc, currentAlert) => {
      const {
        start_time: startTime,
        acknowledged_time: acknowledgedTime,
        end_time: currentEndTime,
        state,
        alert_history = [],
      } = currentAlert._source;
      acc.push(
        ...dataPointsGenerator(
          {
            startTime,
            acknowledgedTime,
            endTime: currentEndTime,
            lastEndTime,
            windowStartTime,
            windowEndTime,
            meta: {
              startTime,
              acknowledgedTime,
              endTime: currentEndTime,
              state,
              errorsCount: alert_history.length,
            },
          },
          lastEndTime
        )
      );
      lastEndTime = currentEndTime;
      return acc;
    }, []);

    // Handle last data point if required.
    const lastAlertDataPoint = [];
    if (lastEndTime && lastEndTime < windowEndTime) {
      lastAlertDataPoint.push({
        x0: lastEndTime,
        x: windowEndTime,
        state: HistoryConstants.TIME_SERIES_ALERT_STATE.NO_ALERTS,
        meta: {
          startTime: lastEndTime,
          endTime: windowEndTime,
        },
      });
    }
    return [...firstAlertDataPoints, ...restAlertsDataPoints, ...lastAlertDataPoint];
  };

  getWindowSize = (poiData, intervalDuration) => {
    /*
      brushAreaStart Duration is to defined the start point for smaller window over zoomer
      on which time line will be displayed. Duration will default to what interval has been
      computed to ES bucket in case it is too small,
      just scale it so that customer can have better experience
    */
    const {
      poiTimeWindow: { endTime },
    } = this.state;
    const brushAreaStartDuration =
      intervalDuration.asMinutes() < 10
        ? moment.duration(HistoryConstants.MIN_HIGHLIGHT_WINDOW_DURATION, 'm')
        : intervalDuration;
    return {
      startTime: endTime.valueOf() - brushAreaStartDuration.asMilliseconds(),
      endTime: endTime.valueOf(),
    };
  };

  getPOIData = async () => {
    const { httpClient, monitorId } = this.props;
    const { poiTimeWindow } = this.state;
    const intervalDuration = calculateInterval(
      moment.duration(poiTimeWindow.endTime - poiTimeWindow.startTime, 'ms')
    );

    try {
      const resp = await httpClient.post('../api/alerting/_search', {
        query: getPOISearchQuery(
          monitorId,
          poiTimeWindow.startTime.valueOf(),
          poiTimeWindow.endTime.valueOf(),
          intervalDuration
        ),
        index: INDEX.ALL_ALERTS,
      });
      if (resp.data.ok) {
        const poiData = get(resp, 'data.resp.aggregations.alerts_over_time.buckets', []).map(
          item => ({
            x: item.key,
            y: item.doc_count,
          })
        );
        this.setState(
          {
            poiData,
            maxAlerts: get(resp, 'data.resp.aggregations.max_alerts.value', 0),
            timeSeriesWindow: {
              ...this.getWindowSize(poiData, intervalDuration),
            },
          },
          () => this.getAlerts()
        );
      } else {
        const parsedError = JSON.parse(resp.data.resp.response);
        this.setState({ queryResponse: parsedError });
      }
    } catch (err) {
      //TODO:: Handle Errors Gracefully.
      console.log('err', err);
    }
  };

  getAlerts = async () => {
    this.setState({
      isLoading: true,
    });
    const { timeSeriesWindow } = this.state;
    const { httpClient, triggers, monitorId } = this.props;
    // Shall we convert this into single call and get all the data ?
    const promises = triggers.map(trigger =>
      httpClient.post('../api/alerting/_search', {
        query: getTimeSeriesSearchRequest(
          monitorId,
          trigger.id,
          timeSeriesWindow.startTime,
          timeSeriesWindow.endTime
        ),
        index: INDEX.ALL_ALERTS,
        size: HistoryConstants.MAX_DOC_COUNT_FOR_ALERTS,
      })
    );
    try {
      //TODO:: If anyone of the call fails, this will Promise will reject automatically.
      // Should we handle each trigger individually and then plot it if we get data ?
      const triggersResponse = await Promise.all(promises);
      const triggersData = triggersResponse.reduce((acc, triggerData, index) => {
        const trigger = triggers[index];
        if (triggerData.data.ok) {
          return {
            ...acc,
            [trigger.id]: this.generatePlotData(get(triggerData, 'data.resp.hits.hits', [])),
          };
        }
      }, {});

      this.setState({
        isLoading: false,
        triggersData,
        prevTimeSeriesWindow: timeSeriesWindow,
      });
    } catch (err) {
      //TODO:: Handle Errors Gracefully.
      console.log('err', err);
    }
  };

  handleDragEnd = area => {
    const { timeSeriesWindow } = this.state;
    if ((area && area.left.getTime() === timeSeriesWindow.startTime) || this.state.isLoading)
      return;
    this.setState(
      {
        prevTimeSeriesWindow: timeSeriesWindow,
        timeSeriesWindow: {
          startTime: area.left.getTime(),
          endTime: area.right.getTime(),
        },
      },
      () => this.getAlerts()
    );
  };
  render() {
    const {
      poiData,
      timeSeriesWindow,
      isLoading,
      triggersData,
      poiTimeWindow,
      maxAlerts,
      prevTimeSeriesWindow,
    } = this.state;
    const { triggers, onShowTrigger } = this.props;
    return (
      <ContentPanel
        title="History"
        titleSize="s"
        bodyStyles={{ minHeight: 200, padding: 0 }}
        actions={[
          <DateRangePicker
            initialStartTime={this.initialStartTime}
            initialEndTime={this.initialEndTime}
            onRangeChange={this.handleRangeChange}
          />,
        ]}
      >
        {triggers.length > 0 ? (
          <React.Fragment>
            <TriggersTimeSeries
              triggers={triggers}
              isLoading={isLoading}
              triggersData={triggersData}
              domainBounds={prevTimeSeriesWindow || timeSeriesWindow}
            />
            <POIChart
              isLoading={isLoading}
              data={poiData}
              onDragStart={() => this.setState({ loading: false })}
              highlightedArea={timeSeriesWindow}
              onDragEnd={this.handleDragEnd}
              xDomain={[poiTimeWindow.startTime, poiTimeWindow.endTime]}
              yDomain={[
                0,
                maxAlerts <= HistoryConstants.MIN_POI_Y_SCALE
                  ? HistoryConstants.MIN_POI_Y_SCALE
                  : maxAlerts,
              ]}
              isDarkMode={this.isDarkMode}
            />
            <EuiHorizontalRule margin="xs" />
            <Legend />
          </React.Fragment>
        ) : (
          <EmptyHistory onShowTrigger={onShowTrigger} />
        )}
      </ContentPanel>
    );
  }
}

MonitorHistory.propTypes = {
  triggers: PropTypes.array.isRequired,
  onShowTrigger: PropTypes.func.isRequired,
};

export default MonitorHistory;
