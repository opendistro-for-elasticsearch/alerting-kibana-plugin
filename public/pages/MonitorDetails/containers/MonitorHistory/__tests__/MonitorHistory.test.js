/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import moment from 'moment-timezone';
import { set } from 'lodash';
import { mount } from 'enzyme';
import { getPOIResponse, getAlertsResponse } from './testHelpers';
import MonitorHistory from '../MonitorHistory';
import { coreMock } from '../../../../../../test/mocks';

moment.tz.setDefault('America/Los_Angeles');

describe('<MonitorHistory/>', () => {
  const initialStartTime = moment('2018-10-15T09:00:00');
  const triggers = [
    {
      name: 'Trigger 1',
      id: '1',
    },
    {
      name: 'Trigger 2',
      id: '2',
    },
  ];
  const httpClient = {
    post: jest.fn(),
    get: jest.fn(),
  };
  beforeEach(() => {
    jest.resetAllMocks();
  });
  test('should show <EmptyHistory/> if no triggers found', (done) => {
    const wrapper = mount(
      <MonitorHistory
        httpClient={httpClient}
        monitorId={'123'}
        onShowTrigger={jest.fn()}
        triggers={[]}
        core={coreMock}
      />
    );
    process.nextTick(() => {
      wrapper.update();
      expect(httpClient.post).toHaveBeenCalledTimes(0);
      expect(wrapper.find('EmptyHistory').length).toBe(1);
      done();
    });
  });
  test('should execute getPOIData, getAlerts on componentDidMount', (done) => {
    const poiResponse = getPOIResponse(initialStartTime);
    httpClient.post.mockResolvedValueOnce(poiResponse);
    httpClient.get.mockResolvedValue({
      ok: true,
      alerts: [],
    });
    const wrapper = mount(
      <MonitorHistory
        httpClient={httpClient}
        monitorId={'123'}
        onShowTrigger={jest.fn()}
        triggers={triggers}
        core={coreMock}
      />
    );
    process.nextTick(() => {
      wrapper.update();
      expect(wrapper.state().poiData).toEqual(
        poiResponse.resp.aggregations.alerts_over_time.buckets.map((item) => ({
          x: item.key,
          y: item.doc_count,
        }))
      );
      expect(httpClient.post).toHaveBeenCalledTimes(1);
      expect(httpClient.get).toHaveBeenCalledTimes(1);
      expect(wrapper.state().maxAlerts).toBe(poiResponse.resp.aggregations.max_alerts.value);
      const triggersData = wrapper.state().triggersData;
      const triggersDataKeys = Object.keys(triggersData);
      expect(triggersDataKeys.length).toBe(2);
      // Validate Ids
      expect(triggersDataKeys[0]).toBe('1');
      expect(triggersDataKeys[1]).toBe('2');
      // Should be not alerting state
      expect(triggersData[triggersDataKeys[0]].length).toBe(1);
      expect(triggersData[triggersDataKeys[1]].length).toBe(1);
      done();
    });
  });
  test('should get 60 mins highlight windowSize', (done) => {
    const poiResponse = getPOIResponse(initialStartTime);
    httpClient.post.mockResolvedValueOnce(poiResponse);
    httpClient.get.mockResolvedValue({
      ok: true,
      alerts: [],
    });
    const wrapper = mount(
      <MonitorHistory
        httpClient={httpClient}
        monitorId={'123'}
        onShowTrigger={jest.fn()}
        triggers={triggers}
        core={coreMock}
      />
    );
    process.nextTick(() => {
      wrapper.update();
      const timeSeriesWindow = wrapper.state().timeSeriesWindow;
      expect(
        moment.duration(timeSeriesWindow.endTime - timeSeriesWindow.startTime).asMinutes()
      ).toBe(60);
      done();
    });
  });
  test('should create appropriate alerts data', (done) => {
    const poiResponse = getPOIResponse(initialStartTime);
    Date.now = jest.fn(() => 1539619200000);
    const alerts = triggers.map((trigger) => [
      getAlertsResponse(
        trigger.id,
        trigger.name,
        '123',
        'Hello',
        moment('2018-10-15T09:00:00').subtract(1, 'h')
      ),
    ]);
    httpClient.post.mockResolvedValueOnce(poiResponse);
    httpClient.get.mockResolvedValue({
      ok: true,
      alerts: alerts,
    });
    const wrapper = mount(
      <MonitorHistory
        httpClient={httpClient}
        monitorId={'123'}
        onShowTrigger={jest.fn()}
        triggers={triggers}
        core={coreMock}
      />
    );
    process.nextTick(() => {
      wrapper.update();
      const triggersData = wrapper.state().triggersData;
      expect(triggersData).toMatchSnapshot();

      done();
    });
  });
  test('should fetch new data on timeSeriesWindow change ', (done) => {
    const poiResponse = getPOIResponse(initialStartTime);
    Date.now = jest.fn(() => 1539619200000);
    httpClient.post.mockResolvedValue({ ok: true }).mockResolvedValueOnce(poiResponse);
    httpClient.get.mockResolvedValue({
      ok: true,
      alerts: [],
    });
    const wrapper = mount(
      <MonitorHistory
        httpClient={httpClient}
        monitorId={'123'}
        onShowTrigger={jest.fn()}
        triggers={triggers}
        core={coreMock}
      />
    );

    const getAlertsSpy = jest.spyOn(wrapper.instance(), 'getAlerts');

    process.nextTick(() => {
      const draggedWindow = {
        startWindow: moment('2018-10-14T09:00:00'),
        endWindow: moment('2018-10-14T10:00:00'),
      };
      // Clear previous calls
      jest.resetAllMocks();
      wrapper.instance().handleDragEnd({
        left: draggedWindow.startWindow.toDate(),
        right: draggedWindow.startWindow.toDate(),
      });
      wrapper.update();
      expect(wrapper.state().timeSeriesWindow).toMatchSnapshot();
      expect(getAlertsSpy).toHaveBeenCalledTimes(1);
      done();
    });
  });
  test('should fall back to max scale if the max alerts are lower than threshold ', (done) => {
    const poiResponse = getPOIResponse(initialStartTime);
    set(poiResponse, 'resp.aggregations.max_alerts.value', 3);

    Date.now = jest.fn(() => 1539619200000);
    httpClient.post.mockResolvedValue({ ok: true }).mockResolvedValueOnce(poiResponse);
    httpClient.get.mockResolvedValue({
      ok: true,
      alerts: [],
    });
    const wrapper = mount(
      <MonitorHistory
        httpClient={httpClient}
        monitorId={'123'}
        onShowTrigger={jest.fn()}
        triggers={triggers}
        core={coreMock}
      />
    );
    process.nextTick(() => {
      wrapper.update();
      expect(wrapper.find('POIChart').props().yDomain).toEqual([0, 5]);
      done();
    });
  });
});
