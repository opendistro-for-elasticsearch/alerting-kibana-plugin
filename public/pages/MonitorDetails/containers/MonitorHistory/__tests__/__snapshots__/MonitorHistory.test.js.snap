// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonitorHistory/> should create appropriate alerts data 1`] = `
Object {
  "1": Array [
    Object {
      "meta": Object {
        "endTime": 1539619200000,
        "startTime": 1539615600000,
      },
      "state": "NO_ALERTS",
      "x": 1539619200000,
      "x0": 1539615600000,
    },
  ],
  "2": Array [
    Object {
      "meta": Object {
        "endTime": 1539619200000,
        "startTime": 1539615600000,
      },
      "state": "NO_ALERTS",
      "x": 1539619200000,
      "x0": 1539615600000,
    },
  ],
}
`;

exports[`<MonitorHistory/> should fetch new data on timeSeriesWindow change  1`] = `
Object {
  "endTime": 1539532800000,
  "startTime": 1539532800000,
}
`;
